Composer/packages/client/src/store/reducer/index.ts

```diff
@@ -417,14 +417,25 @@ const setUserSessionExpired: ReducerFunc = (state, { expired } = {}) => {
   return state;
 };
 
-const setPublishTypes: ReducerFunc = (state, { response }) => {
-  state.publishTypes = response;
+const setPublishTypes: ReducerFunc = (state, { typelist }) => {
+  // const types: string[] = [];
+  // typelist.map(item => types.push(item.name));
+  state.publishTypes = typelist;
   return state;
 };
 
 const publishSuccess: ReducerFunc = (state, payload) => {
-  state.botEndpoints[state.projectId] = `${payload.results?.result?.endpoint || 'http://localhost:3979'}/api/messages`;
-  state.botStatus = BotStatus.connected;
+  if (payload.target.name === 'default' && payload.endpointURL) {
+    state.botEndpoints[state.projectId] = `${payload.endpointURL || 'http://localhost:3979'}/api/messages`;
```

Comment:
I'm not sure we can count on 'http://localhost:3979' any longer now that the runtime will find and use an available port.

Reply:
Removed it